crates/provider/src/native.rs

```diff
@@ -296,19 +299,77 @@ impl<T: FileSystem + Send + Sync> Provider for NativeProvider<T> {
         Ok(format!("tail -f {}/{}.log", self.tmp_dir, name))
     }
 
+    // TODO: Add test
     async fn pause(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process = self.get_node_from_name(node_name)?;
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -STOP {}", process.pid)],
+                NativeRunCommandOptions {
+                    is_failure_allowed: true,
+                },
+            )
+            .await?;
         Ok(())
     }
 
+    // TODO: Add test
     async fn resume(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process = self.get_node_from_name(node_name)?;
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -CONT {}", process.pid)],
+                NativeRunCommandOptions {
+                    is_failure_allowed: true,
+                },
+            )
+            .await?;
         Ok(())
     }
 
-    async fn restart(&mut self, node_name: &str, after_secs: u16) -> Result<bool, ProviderError> {
-        // TODO: impl using run_command
-        // use &mut here since we mostly need to update the the pid/maps
+    // TODO: Add test
+    async fn restart(
+        &mut self,
+        node_name: &str,
+        after_secs: Option<u16>,
+    ) -> Result<bool, ProviderError> {
+        let process = self.get_node_from_name(node_name)?;
+
+        self.run_command(
+            vec![format!("kill -9 {:?}", process.pid)],
+            NativeRunCommandOptions {
+                is_failure_allowed: true,
+            },
+        )
+        .await?;
+
+        if let Some(secs) = after_secs {
+            sleep(Duration::from_secs(secs.into())).await;
+        }
+
+        let process: &mut Process =
+            self.process_map
+                .get_mut(node_name)
+                .ok_or(ProviderError::MissingNodeInfo(
+                    node_name.to_owned(),
+                    "process".into(),
+                ))?;
+
+        let mapped_env: HashMap<String, String> = env::vars().map(|(k, v)| (k, v)).collect();
```

Comment:
The idea here is to add `our` env vars to the process, we should get those from the original process (if were set).

Reply:
Actually, this will require a little more of work since we are not even storing the original `ProcessEnvironment` in the map, let commented out for now and I will add in a follow-up pr.